examples/spark-pi-default.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2017 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+apiVersion: "sparkoperator.k8s.io/v1alpha1"
+kind: SparkApplication
+metadata:
+  name: spark-pi
+  namespace: default
+spec:
+  type: Scala
+  mode: cluster
+  image: Default
```

Comment:
@liyinan926 should I add this to all examples as default, and rename this yaml file as `spark-pi-custom` ? Or leave it as it is?

Reply:
What if we simply make `image` optional, and if it's not specified, use the default image, instead of using this `Default` value?

Reply:
It is possible, but would require getting rid of this - https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/sparkctl/cmd/create.go#L138 . Sparkctl validates it. Thoughts?

Reply:
I think we can get rid of that check for now.